packages/sdk/src/cbor.ts

```diff
@@ -0,0 +1,67 @@
+import { decode } from 'cbor2';
+
+import { TokenAmount } from './plt/types.js';
+import * as AccountAddress from './types/AccountAddress.js';
+
+/**
+ * Register CBOR encoders for all types at the top level so they are globally available in the application.
+ * This is safe to do as multiple encoders for the same CBOR tag is not a problem.
+ *
+ * Currently, this auto-registers the following encoders:
+ * - `AccountAddress`: For encoding Concordium account addresses in CBOR format
+ * - `TokenAmount`: For encoding protocol-level token amounts in CBOR format
+ */
+AccountAddress.registerCBOREncoder();
+TokenAmount.registerCBOREncoder();
```

Comment:
Loading this module now has side-effects, meaning bundlers cannot just leave it out for code elimination.
So I think we need to either list it in the `package.json` `"sideEffects": [...]` or maybe not do side effects and move these into `registerCBORDecoders`

Reply:
Yes I know, good point with regards to the `"sideEffects"` field. Maybe it's too agressive to do this here, but the intention is to provide at an easy way to only do this (i.e. register type-specific encoders) once in your application. 

Could do something like put all the encoder registrations into `@concordium/web-sdk/extensions/cbor2` and then put the rest into `@concordium/web-sdk/types`? Then the side effects would be limited to cases where users need to cbor encode our types.

`cbor2` (and most other cbor js libraries) works by globally registering encoders/decoders.